packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -264,90 +97,80 @@ const GroupPropertyAction = ({
   keySet,
   returnFn,
 }: GroupPropertyActionProps) => {
-  const iModelConnection = useActiveIModelConnection() as IModelConnection;
+  const iModelConnection = useActiveIModelConnection();
   const { getAccessToken } = useGroupingMappingApiConfig();
   const mappingClient = useMappingClient();
   const [propertyName, setPropertyName] = useState<string>("");
   const [dataType, setDataType] = useState<DataType>(DataType.Undefined);
   const [quantityType, setQuantityType] = useState<QuantityType>(QuantityType.Undefined);
-  const [classToPropertiesMapping, setClassToPropertiesMapping] =
-    useState<Map<string, Property[]>>();
-  const [ecProperties, setEcProperties] = useState<ECProperty[]>(
-    []
-  );
+  const [selectedProperties, setSelectedProperties] = useState<PropertyMetaData[]>([]);
+  const [propertiesMetaData, setPropertiesMetaData] = useState<PropertyMetaData[]>([]);
+  const [propertiesNotFoundAlert, setPropertiesNotFoundAlert] = useState<boolean>(false);
   const [validator, showValidationMessage] = useValidator();
-  const [propertyAlert, setPropertyAlert] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchInput, setSearchInput] = useState<string>("");
+  const [activeSearchInput, setActiveSearchInput] = useState<string>("");
+  const [searched, setSearched] = useState<boolean>(false);
+  const [activeDragProperty, setActiveDragProperty] = useState<PropertyMetaData>();
```

Comment:
No default provided for the state which cannot be `undefined`. Change the type or provide a default.

Reply:
Changed the type. I figured that not passing anything was implicit enough.